### PR TITLE
return region error when TiKV is closing

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**
Signed-off-by: zhangjinpeng1987 <zhangjinpeng@pingcap.com>

## What have you changed? (mandatory)

Return an RegionError when TiKV is closing to tell the client that this region is unavailable temporarily, the client should retry the request in other TiKVs.

## What are the type of the changes? (mandatory)

- Improvement (change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

- [ ] killing test

## Does this PR affect documentation (docs) or release note? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)

No

### CODE REVIEW

File: src/server/service/kv.rs [hunk 1]
```diff
@@ -1824,6 +1824,13 @@ fn extract_region_error<T>(res: &storage::Result<T>) -> Option<RegionError> {
             err.set_server_is_busy(server_is_busy_err);
             Some(err)
         }
+        Err(storage::Error::Closed) => {
```
Comment (siddontang) [inline src/server/service/kv.rs]: any test to cover this?

Comment (AndreMouche) [reply src/server/service/kv.rs]: `Error::Closed` is enough?

Comment (zhangjinpeng1987) [reply src/server/service/kv.rs]: We have a long time running schrodinger test to cover this scenario.

**Comment by zhangjinpeng1987** [comment]
@disksing @AndreMouche PTAL again.

**Pull request merged**